### ISSUE TITLE
Meeting on Wednesday 22nd - 17:00 GMT

### ISSUE THREAD

**Issue by pepibumur** [tuist/tuist#869]
I'm opening this issue to include the points that we'd like to discuss in our next sync call. Feel free to update the description of this issue to add more points:

- General intro / welcome new joiners
- Tuist usage updates (a round the table update of how everyone is getting along with Tuist / any issues / blockers)
- @kwridan: Performance / Optimising generation (proposal: https://github.com/tuist/tuist/issues/820)
- @ollieatkinson: Always good to talk over any open P/R's and how to unblock them.
- @ollieatkinson: There's also been some hesitation on my side in regards to starting new work knowing we are thinking of moving towards a reactive architecture - are we still considering this and should we start prepping some RFC documents
- @pepibumur - Release cadence and captains.
- @pepibumur  - Slack vs GitHub
- Open Issues:
  - #889: API Stability Guidelines

**Comment by pepibumur** [comment]
Here are the notes from the meeting:

# Notes from Tuist's meeting

## Introductions
We all introduce ourselves briefly explaining who we are and at which company we work.

## Usage updates
Ollie:
- They upgraded to 1.1
- They haven't increased the number of modules.
- They are not using project helpers yet.
- They are still looking into how they might be able to improve build times on CI.
- They'd like to build from source locally.
- How can they debug issues with the tool? They are interested in adding verbose logging.
- He opened a PR with an intermediate solution.
- He's going to open up another PR with a more generic approach.

Andrea & Donald:
- They started 2 months.
- They generate 23 frameworks with Tuist.
- They opened the last PR to generate the app.
- They are extensively using the project helpers.
- The main problems are:
- It takes some time to generate the projects.
- In the process of generating the projects they learned a lot of things. They'd like to share some scripts and ideas on how to port projects.

Vytis:
- He is interested in using Tuist at the company
- They are creating a proof of concept with around 10 modules.
- The main reason is that the setup of the module is very ad-hoc and there are many hacks around.
- The main rational is cleaning everything up.
- He's also learning 
- He's focusing on porting the frameworks at the moment.

Lakpa:
- He is using Tuist in his personal project.
- The company is using XcodeGen.
- He likes the way Tuist was setup.
- He'd like to add some benchmarking to the project.
- We don't know how much improvement we are introducing in regards to performance.
- His setup is very small so he's not encountering some performance issues.
- He found an option that we can maybe use to know where the time is being spent.

Kas:
- They generate the application but they took a shortcut.
- They use Tuist as a library. They wrote a library on top of Tuist.
- They have the majority of projects generated.
- The biggest project to migrate was the main app.
- It's worth using the latest Tuist master.
- He suggests using the latest version from master because there was a bug that caused some headers to be copied to the framework bundle.
- Was: They should use --project-only to generate only the project for the module that is in the current directory.

## Performance
- Vytis: If we take a large project, we should be able to see with instruments where's the performance bottleneck.
- Pedro: We should document best practices about how to architect & structure the project.
- Kas:
	- They built a tool to generate a large project that we can use for benchmarking.
	- They noticed that the usage of precompiled nodes adds up to the bottleneck.
- Ollie:
	- We can use Docker images to make sure that we always use the same amount of resources.
	- We'd need to make some changes to enable the Linux support.
- Kas:
	- For the purpose of the benchmarking, we can check out master and the branch, and then compare
- Vytis:
	- That way we don't need to hold the state anywhere.
- Kas:
	- If each PR adds up a 0.1%, over time we might end up with a lot of increase.
- Lapka:
	- Since in Tuist we have a lot of layers, would simplify the layers help?
	- In some places we do iterations multiple times. 

## Talk over any open PRs
- He'd like to call out that some PRs have been opened for a long time.

## How do we work?
- Should we introduce patterns?
- Are there patterns that will allow us to work independently from each other?
- We should start writing down a governance. Basically describe the way we work.
- We don't need to go with specific details. It's more 

## Release cadence and captains
Pedro suggests rotating the release captain role every 2 weeks. He'll include a list of people for release captains and on the documentation website we'll mention who will be the release captain of the next release.

## Slack vs GitHub
- Pedro suggests encouraging people to open issue instead of sorting out issues on Slack for visibility reasons.

## Breaking changes
- Vytis will document what are breaking changes.
- He'd like to research how other projects do it.
